planet/src/main/webapp/js/profile.js

```diff
@@ -100,30 +101,35 @@ function switchSection(linkId, sectionId) {
     diplaySection(sectionId);
 }
 
-async function loadUserInformation(username) {
+async function loadUserInformation(username, blobKey) {
     const usersReference = database.ref('/users');
     const userSnapshot = await usersReference.orderByChild('username').equalTo(username).once('value');
-    if (userSnapshot.val() === null) {
-        document.getElementById('profile-page').remove();
-        document.getElementById('not-found-message').style.display = 'block';
-        return;
-    }
     if (userSnapshot.val() !== null) {
         // The userSnapshot.val() contains one property (user)
         for (const property in userSnapshot.val()) {
             userId = property;
             user = userSnapshot.val()[property];
         }
-    
-        if (user['emailVerified']) {
+        // If the blobkey exists, get the image url and store in the database
+        if (blobKey !== 'null') {
+            let picture = await fetch('/serve?blob-key=' + blobKey);
+            database.ref('/users/' + userId).update({
+                profilePic: picture.url
+            })
+            .then(() => {
+                window.location.href = '/user/' + username;
```

Comment:
Why do we need to redirect after this DB write?

Reply:
Actually, it's just to hide the blobkey from the user. I don't know if it's important but if we don't redirect the user can see the blobkey from the profile's URL. Do you think I should remove it?

Reply:
What does the blob key give the user access to? Does it let them just download their own profile pic? If so, it should be fine to leave it.

Another consideration would be whether or not this is noticeable for the user. Does this line cause the whole page to reload or does the browser just change the URL bar but not reload anything?

Reply:
It actually reloads the page that's why I didn't like this solution much. As for the info in the blobkey, it's just the blobkey of the user's profile picture. I will remove it then and keep the URL with the blobkey!